.github/workflows/step_build.yml

```diff
@@ -0,0 +1,58 @@
+name: build
+run-name: Build test
+
+on:
+  workflow_call:
+    inputs:
+      upload:
+        type: boolean
+        required: false
+        default: false
+        description: Upload build artifacts
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-build
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+      - name: Build package
+        run: |
+          python -m pip install build wheel
+
+          # NOTE: These builds and verifications of the builds can be done more
+          # robustily with jupyter-releaser.
+          #
+          # Removed the check on size of package as we are distributing tests/ with
+          # sdist now and they are around 8MB. Seems like original check was to make
+          # sure we are not distributing node_modules and we are quite safe with that
+          # with hatch build system.
+          #
+          # Build jupytext package
+          python -m build
+
+          # Build lab extension(s)
+          npm pack --pack-destination dist jupyterlab/packages/*
```

Comment:
Is this redundant with `python -m build`? I mean, I guess this step is now addressed by `hatch` so I would expect that we can remove this?

Reply:
`hatch`'s build hook `jupyter-builder` builds the lab extension. When I say building, it transpiles the TS to mimified jS. But it does not run packaging step. After all we need to use `npm pack` before we can publish it to `npm` registry and that is the step we are checking here. Does it make sense?

Reply:
Yes it does! The main reason I am asking is that I though that installing JupyterLab extensions from npm had become deprecated, so I am not sure I want to expose this file, but I will do as you recommend.

Reply:
Yes, you are right. The default extension manager now is [PyPI](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#managing-extensions-using-the-extension-manager). But installing extensions from `npm` is still [supported](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#un-installing-using-jupyter-labextension) even though it is not very convenient way. More over `jupyter-releaser` workflows (used by most of JupyterLab extension ecosystem) publishes them to `npm`. As you have been already doing it, you can continue publishing them just to be consistent.